compiler/IREmitter/IREmitter.cc

```diff
@@ -751,6 +751,10 @@ void emitUserBody(CompilerState &base, cfg::CFG &cfg, const IREmitterContext &ir
                         auto description = fmt::format("Parameter '{}'", bind.bind.variable.toString(cs, cfg));
                         if (argInfo.flags.isBlock) {
                             IREmitterHelpers::emitTypeTestForBlock(cs, builder, var, expectedType, description);
+                        } else if (argInfo.flags.isRepeated && !argInfo.flags.isKeyword) {
+                            // Signature types for rest arguments apply to each individual element of
+                            // the rest arg, not the actual argument itself.
+                            IREmitterHelpers::emitTypeTestForRestArg(cs, builder, var, expectedType, description);
```

Comment:
Do we have the same problem with `**kwargs`?

For those, sorbet-runtime will check that for each keyword arg, the value has the mentioned type

```ruby
sig {params(kwargs: Integer).void}
def foo(**kwargs)
end

foo(a: 1, b: 2, c: 'nope')
```

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7Bparams%28kwargs%3A%20Integer%29.void%7D%0Adef%20foo%28**kwargs%29%0Aend%0A%0Afoo%28a%3A%201%2C%20b%3A%202%2C%20c%3A%20'nope'%29)

Reply:
I haven't tested it, but I very much imagine that we do.  I can write a disabled testcase for this tomorrow.  I think fixing the kwsplat case is a little more complicated because iterating over a hash table would require a separate function that the typecheck needs to live in.

(Also not sure how many people write types on their kwsplat arguments.)

Reply:
Testcase in #4860.